File: algorithm/acwing/2023WinterVacationDayOne/algorithm/src/main/java/week3/day3_D/main.py
import math

n = int(input())
w = list(map(int, input().split()))

s = sum(w) / n
ss = 0
for i in range(n):
    ss += (s - w[i]) ** 2
ss /= n

sss = ss ** (1 / 2)

for i in range(n):
    print((w[i] - s) / sss)

File: algorithm/acwing/2023WinterVacationDayOne/algorithm/src/main/java/week3/day3_D/Main.java
package week3.day3_D;

import java.util.Scanner;

public class Main {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int n = sc.nextInt();
        int a[] = new int[n + 1];
        double sum = 0;
        for (int i = 1; i <= n; i++) {
            a[i] = sc.nextInt();
            sum += a[i];
        }
        double a_average = (double) sum / n;
        double da = 0;
        for (int i = 1; i <= n; i++) {
            da += (a[i] - a_average) * (a[i] - a_average);
        }
        da = da / n;
        da = Math.sqrt(da);
        //double b[]=new double[n];
        for (int i = 1; i <= n; i++) {
            System.out.println((a[i] - a_average) / da);
        }
    }
}
